File: web/src/api/challenge.ts
import { get, post, put, del } from './client'
import type { Challenge, Difficulty } from '@/types'

export interface ChallengeParams {
  name: string
  description?: string
  image_name: string
  image_tag?: string
  difficulty: Difficulty
  base_score: number
  exposed_ports?: string  // e.g. "80,8080"
  cpu_limit?: number
  mem_limit?: number
}

export const challengeApi = {
  // 获取比赛的题目列表
  list: (gameId: number) => get<Challenge[]>(`/games/${gameId}/challenges`),
  
  // 添加题目到比赛
  create: (gameId: number, data: ChallengeParams) => post<Challenge>(`/games/${gameId}/challenges`, data),
  
  // 更新题目
  update: (gameId: number, challengeId: number, data: Partial<ChallengeParams>) => 
    put<Challenge>(`/games/${gameId}/challenges/${challengeId}`, data),
  
  // 删除题目
  delete: (gameId: number, challengeId: number) => 
    del<void>(`/games/${gameId}/challenges/${challengeId}`),
}

File: internal/model/challenge.go
package model

import "time"

type Challenge struct {
    ID           int64     `json:"id" gorm:"primaryKey"`
    GameID       int64     `json:"game_id" gorm:"index"`
    Name         string    `json:"name"`
    Description  string    `json:"description"`
    ImageName    string    `json:"image_name"`
    ImageTag     string    `json:"image_tag" gorm:"default:latest"`
    Difficulty   string    `json:"difficulty" gorm:"default:medium"`
    BaseScore    int       `json:"base_score" gorm:"default:100"`
    ExposedPorts string    `json:"exposed_ports"`
    CPULimit     float64   `json:"cpu_limit" gorm:"default:0.5"`
    MemLimit     int       `json:"mem_limit" gorm:"default:256"`
    CreatedAt    time.Time `json:"created_at"`
}
